include/zephyr/location.h

```diff
@@ -0,0 +1,340 @@
+/**
+ * @file location.h
+ *
+ * @brief Public location API.
+ *
+ */
+
+/*
+ * Copyright (c) 2022, Trackunit A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_LOCATION_H_
+#define ZEPHYR_INCLUDE_LOCATION_H_
+
+/**
+ * @brief Location Interface
+ * @defgroup location_interface Location Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Location data
+ *
+ * The latitude and longitude are in the unit micro degree. 1 micro
+ * degree corresponds to approx. 11cm along the earths surface.
+ *
+ * The accuracy describes the radius within which the real
+ * position is with a 68.2% certainty, in millimeters.
+ */
+struct location_position {
+	/** Latitudal position in microdegrees (0 to +-180000000) */
+	int32_t latitude;
+	/** Longitudal position in microdegrees (0 to +-180000000) */
+	int32_t longitude;
+	/** The uptime when the position was determined */
+	int64_t uptime;
```

Comment:
For NMEA sentences we would be able to get the UTC time, but seems like we have nowhere to store it in this API?

Reply:
This is by design, the real time will get its own API and subsystem, where devices which can provide it will register themselves, as well as timekeepers like RTC clocks. We will create a design proposal for this later.

Reply:
I agree that we should somehow be able to sync the onboard time with the GNSS time, which should be another proposal. But for this `location` API, I would also need to know the time of the fix, which doesn't seem possible?